src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -8789,9 +8789,33 @@ we will have to use something else.
                         tmp = nullptr;
                     }
                     return ;
+                } else if (args.size() > 1) {
+                    throw SemanticError("set accepts only 1 argument for now, got " +
+                                        std::to_string(args.size()) + " arguments instead.",
+                                        x.base.base.loc);
                 }
-
-                throw SemanticError("set is only used for an empty set for now.", x.base.base.loc);
+                if ( assign_asr_target == nullptr ) {
+                    throw SemanticError("set from list cannot be called without target type for now", x.base.base.loc);
+                }
+                ASR::expr_t *arg = args[0].m_value; 
+                ASR::ttype_t *type = ASRUtils::expr_type(arg);
+                if(!ASR::is_a<ASR::ListConstant_t>(*arg)) {
+                    throw SemanticError("set accepts only list for now, got " +
+                                        ASRUtils::type_to_str(type) + " type.", x.base.base.loc);
```

Comment:
On a second look, I think we should mention clearly:
```suggestion
                    throw SemanticError("set accepts only list constant for now, got " +
                                        ASRUtils::type_to_str(type) + " type.", x.base.base.loc);
```